x/tokenomics/keeper/token_logic_modules.go

```diff
@@ -557,13 +655,45 @@ func (k Keeper) ensureClaimAmountLimits(
 	// settlement. Relay miners use the appStake at the beginning of a session to determine
 	// the maximum amount they can claim. We need to somehow access and propagate this
 	// value (via context?) so it is the same for all TLM processors for each claim.
-	// Note that this will also need to incorporate MintPerClaimGlobalInflation because
-	// applications are being overcharged by that amount in the meantime. Whatever the
-	// solution and implementation ends up being, make sure to KISS.
+	// Note that this also incorporates MintPerClaimGlobalInflation since applications
+	// are being overcharged by that amount and the funds are sent to the DAO/PNF
+	// before being reimbursed to the application in the future.
 	appStake := application.GetStake()
 
+	// TODO_BETA(red-0ne): Make sure that the relay miner logic also accounts for this
+	// when deciding to serve an application.
+
+	// The application should have enough stake to cover for the global mint reimbursement.
+	// This amount is deducted from the maximum claimable amount.
+	globalInflationCoin, _ := calculateGlobalPerClaimMintInflationFromSettlementAmount(claimSettlementCoin)
+
+	// TODO_BETA(@red-0ne): Introduce a session sliding window to account for potential consumption
```

Comment:
Looks like something you'll do in MainNet?

Reply:
This is important, it's in next week's tasks.

Reply:
I'm skipping this long comment but will take a look next week

Reply:
Implemented in #895